controllers/vcdmachine_controller.go

```diff
@@ -1049,6 +1085,47 @@ func (r *VCDMachineReconciler) reconcileNormal(ctx context.Context, cluster *clu
 	return ctrl.Result{}, nil
 }
 
+// generateNetworkInitializationScriptForIgnition creates the bash script that will create the networkd units stored in metadata
+// and consumed by ignition
+func generateNetworkInitializationScriptForIgnition(networkConnection *types.NetworkConnectionSection, vdcManager *vcdsdk.VdcManager) (string, error) {
+	var networkMetadata strings.Builder
+	networkMetadata.WriteString("#!/bin/sh\n")
+	networkMetadata.WriteString("set -x\n")
+
+	for _, network := range networkConnection.NetworkConnection {
+		unitFile := "/etc/systemd/network/" + network.Network + ".network"
```

Comment:
> do we say that we don't have any customization of the image, and directly use the ignition-based bootstrap?

That is correct. Note that I provided a template of ignition but it can be customized by the user when deploying a cluster.

> We disable some things such as IPv6 in cloud-init

We can [disable IPv6 in ignition](https://www.flatcar.org/docs/latest/setup/customization/network-config-with-networkd/#turn-off-ipv6-on-specific-interfaces) as well to align with cloud-init.

> Can we use a formatted string for this, or a templatized method to improve readability.

That makes sense, we'll look into this.